src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -408,9 +411,12 @@ async def build_pex(
     requirement_count: int
 
     if isinstance(request.requirements, (Lockfile, LockfileContent)):
+        is_monolithic_resolve = True
         resolve_name = request.requirements.resolve_name
+        synthetic_lock = True
         if isinstance(request.requirements, Lockfile):
             lock_path = request.requirements.file_path
+            synthetic_lock = False
```

Comment:
This probably deserves a better name :see_no_evil:

Reply:
It makes sense to me. What's confusing to me is having `synthetic_lock = True` right above the `isinstance` check. Instead, I recommend having it in a parallel position in the `if` and `else` for symmetry, like how `lock_bytes` works.